lib/services/database_service.dart

```diff
@@ -49,10 +61,15 @@ class DatabaseService {
       _firestoreSubscriptions.processingFailures =
           _firestore.connectToProcessingFailures(userId, _storeController);
     } catch (error, trace) {
-      _storeController.add(AddProblem.from(
-          message: error.toString(),
-          traceString: trace.toString(),
-          type: ProblemType.observeProcessingFailures));
+      _storeController.add(
+        AddProblem(
+          problem: ObserveProcessingFailuresProblem.by(
```

Comment:
Yeah this also makes sense to me - at the ObserveProcessingFailures event we can get a ObserveProcessingFailuresProblem. Nice one!